core/src/main/scala/org/apache/spark/scheduler/cluster/SchedulerBackendUtils.scala

```diff
@@ -44,4 +44,12 @@ private[spark] object SchedulerBackendUtils {
       conf.get(EXECUTOR_INSTANCES).getOrElse(numExecutors)
     }
   }
+
+  def getMaxTargetExecutorNumber(conf: SparkConf): Int = {
```

Comment:
https://github.com/apache/spark/blob/9a990b5a40de3c3a17801dd4dbd4f48e3f399815/core/src/main/scala/org/apache/spark/deploy/ExecutorFailureTracker.scala#L86-L95

Is this different from the `effectNumExecutors` in the above code? Don't we need to consider scenarios where `isStreamingDynamicAllocationEnabled` is true?

Reply:
Also, there are a lot places bypassing the minExecutor adjusting